design/fast_finality.md

```diff
@@ -0,0 +1,276 @@
+# Fast finality design
+
+## Scope
+
+The introduction of a single sequencer to improve cost-per-transaction during the bootstrapping phase, as 
+described in the [Bootstrapping Strategy design doc](./Bootstrapping_strategy.md).
+
+## Glossary
+
+* *Light batch*: A set of transactions produced by the designated sequencer, considered to be soft-finalised in that 
+  order
+
+## Requirements
+
+* Finality
+  * Transaction *soft* finality (finality guaranteed by the sequencer) is achieved in under one second
+  * There is eventual transaction *hard* finality (finality guaranteed by the L1)
+  * The sequencer is incentivised to achieve hard finality on an agreed cadence
+  * The sequencer is incentivised to hard-finalise transactions in the same order they are soft-finalised
+  * The sequencer is not able to "rewrite history" (or is strongly disincentivised from doing so), even for soft-final
+    transactions
+* Censorship resistance
+  * End-users can bypass the sequencer to include transactions on the L1 directly (possibly at higher cost and slower 
+    finality)
+  * The sequencer distributes all soft-finalised transactions to all nodes
+  * Nodes can easily prove whether a given soft-finalised transaction was hard-finalised, and in the correct order
+* Value-extraction resistance
+  * The sequencer cannot precompute the effects of running a given set of transactions without committing to that set 
+    of transactions (e.g. by running a single transaction, then `eth_call`ing to see how it has affected a smart 
+    contract's state)
+* Cost
+  * L1 transaction costs can be driven lower at the expense of extending the hard-finality window
+* User/dev experience
+  * The responses to RPC calls reflect the soft-finalised transactions, and not just the hard-finalised transactions
+* Resilience
+  * Failover does not require a governance action
+  * During failover, it is acceptable to break the one-second soft-finality guarantee 
+  * During failover, it is acceptable to drop transactions that have not been soft-finalised
+
+## Assumptions
+
+* There is a single aggregator that is also the genesis node. This node is known as the *sequencer*
+* The enclave has a start-up delay that exceeds the production time for light batches (see below)
+
+## Constraints
+
+* It must be possible to only publish rollups to the L1 every `x` blocks, where `x` >> 1
```

Comment:
This point's a bit hard to read, maybe "must be possible to" should be "should"?

Reply:
I'll just fold these into the requirements.